src/vma/dev/ring_bond.cpp

```diff
@@ -392,17 +395,24 @@ void ring_bond::inc_tx_retransmissions(ring_user_id_t id)
 
 bool ring_bond::reclaim_recv_buffers(descq_t *rx_reuse)
 {
-	devide_buffers_helper(rx_reuse, m_buffer_per_ring);
+	/* use this local array to avoid locking mechanizm
+	 * for threads synchronization. So every thread should use
+	 * own array. Set hardcoded number to meet C++11
+	 * VLA is not an official part of C++11.
+	 */
+	descq_t buffer_per_ring[MAX_NUM_RING_RESOURCES];
+
+	devide_buffers_helper(rx_reuse, buffer_per_ring);
 	for (uint32_t i = 0; i < m_n_num_resources; i++) {
-		if (m_buffer_per_ring[i].size() > 0) {
-			if (!m_bond_rings[i]->reclaim_recv_buffers(&m_buffer_per_ring[i])) {
-				g_buffer_pool_rx->put_buffers_after_deref_thread_safe(&m_buffer_per_ring[i]);
+		if (buffer_per_ring[i].size() > 0) {
+			if (!m_bond_rings[i]->reclaim_recv_buffers(&buffer_per_ring[i])) {
+				g_buffer_pool_rx->put_buffers_after_deref_thread_safe(&buffer_per_ring[i]);
 			}
 		}
 	}
 
-	if (m_buffer_per_ring[m_n_num_resources].size() > 0)
-		g_buffer_pool_rx->put_buffers_after_deref_thread_safe(&m_buffer_per_ring[m_n_num_resources]);
+	if (buffer_per_ring[m_n_num_resources].size() > 0)
+		g_buffer_pool_rx->put_buffers_after_deref_thread_safe(&buffer_per_ring[m_n_num_resources]);
```

Comment:
Here you may have array overflow when accessing:
```
buffer_per_ring[m_n_num_resources]
```
if m_n_num_resources is 10 you are out of range